skimage/measure/_structural_similarity.py

```diff
@@ -48,6 +155,69 @@ def structural_similarity(X, Y, win_size=7,
     if not X.shape == Y.shape:
         raise ValueError('Input images must have the same dimensions.')
 
+    if image_content_weighting and gradient:
+        raise ValueError(
+            "gradient not implemented for image content weighted case")
+
+    # default treats 3D arrays with shape[2] == 3 as multichannel
+    if multichannel is None:
+        if X.ndim == 3 and X.shape[2] == 3:
+            multichannel = True
+        else:
+            multichannel = False
+
+    if multichannel:
+        # loop over channels
+        args = locals()
+        args.pop('X')
+        args.pop('Y')
+        args['multichannel'] = False
```

Comment:
I see that later you want to pass the other arguments. I would say, pass those explicitly, even though it's more of a pain for that one line. It will simplify this part dramatically.